### ISSUE TITLE
403 - Resource not accessible by integration

### ISSUE THREAD

**Issue by dblock** [MeilCli/danger-action#390]
## Description

Coming from https://github.com/hashie/hashie/runs/5692966922?check_suite_focus=true (https://github.com/hashie/hashie/runs/5692966922?check_suite_focus=true) getting a 403 when danger tries to comment on the issue. 

```
`on_complete': POST https://api.github.com/repos/hashie/hashie/issues/561/comments: 403 - Resource not accessible by integration // See: https://docs.github.com/rest/reference/issues#create-an-issue-comment 
```

What's the best way to deal with this?

In other places we used a different token [like so](https://github.com/slack-ruby/slack-ruby-bot-server/blob/master/.github/workflows/danger.yml#L22), but maybe you have a better idea?

**Comment by dblock** [comment]
https://github.com/danger/danger-js/issues/877 looks similar

**Comment by MeilCli** [comment]
@dblock 
GitHub Token is restricited to read only permission when PR from forked repository.

In other case, I faced same problem.(ref: https://github.com/MeilCli/common-lint-reporter/blob/master/documents/oss-or-dependabot-usage.md)
I fixed problem, I use `workflow_run`.

But I don't feel easy to resolve your case. Because `workflow_run` hasn't PullRequest information. So we need pass the information to `workflow_run` workflow from `pull_request` workflow.

Other solution, we can use `pull_request_target`. It dose not restricited permission.
ref: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target
But, we should attent to security!
I know convinient setting that is aprroving workflow run on a pull request from a public fork.
ref: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
